addons/external-dns/gcp/credentials/main.tf

```diff
@@ -0,0 +1,25 @@
+data "terraform_remote_state" "infra" {
+  backend = "local"
+  config = {
+    path = "../../../../infra/gcp/terraform.tfstate.d/gcp-${var.tsb_mp["cluster_id"]}-${var.gcp_k8s_regions[tonumber(var.tsb_mp["cluster_id"])]}/terraform.tfstate"
+  }
+}
+
+locals {
+  dns_project = endswith(var.tsb_fqdn, ".gcp.cx.tetrate.info") ? "dns-terraform-sandbox" : data.terraform_remote_state.infra.outputs.project_id
+}
+
+resource "google_service_account" "external_dns" {
```

Comment:
I think the right place for the "credentials" function it is actual "external_dns" function, however the external dns function should reference the pre-created subzones, i.e. each cluster external dns records should under the cluster name itself, to avoid the naming collisions between the clusters as well as to keep the configuration manageable.

Reply:
I think the right place for the "credentials" function it is actual "dns subzone that will provide not only zone reference but the credentials as well for external dns to use" function, however the external dns function should reference the pre-created subzones, i.e. each cluster external dns records should under the cluster name itself, to avoid the naming collisions between the clusters as well as to keep the configuration manageable.